CI: Release workflow Docker build fails — hatch-vcs has no .git in build context

## Context

Every push to `main` since commit `849566f build: replace static version with hatch-vcs dynamic versioning` fails the **Release** workflow at the *Build and Push Docker Image* step:

```
#24 1.469       LookupError: Error getting the version from source `vcs`: setuptools-scm
##[error]buildx failed: process "/bin/sh -c uv sync --frozen --no-dev" did not complete successfully: exit code: 1
```

## Root cause

[`Dockerfile`](Dockerfile) copies only `pyproject.toml`, `uv.lock*`, `README.md`, and `src/` into the build context — **not the `.git/` directory**. When `uv sync --frozen --no-dev` runs inside the build, it tries to install the local `intervals-icu-mcp` package, which now uses `hatch-vcs` to derive its version from `git describe`. With no git history in the build context, `setuptools-scm`/`hatch-vcs` raises `LookupError`.

## Acceptance criteria

Pick **one** of the following approaches and implement it:

- [ ] **Option A — Build-arg version (recommended).** Add a `--build-arg PKG_VERSION=...` to `release.yml` computed via `git describe --tags --dirty --always` in a workflow step. In the `Dockerfile`, accept it as `ARG PKG_VERSION` and set `ENV SETUPTOOLS_SCM_PRETEND_VERSION_FOR_INTERVALS_ICU_MCP=$PKG_VERSION` before `uv sync`. Smallest diff, no image bloat.
- [ ] **Option B — Copy `.git/` into build context.** Add `COPY .git/ ./.git/` to the `Dockerfile`. Simple but bloats the layer cache and ships git history into the image (can be mitigated with multi-stage that drops it before final stage).
- [ ] **Option C — External wheel build.** Build the wheel outside Docker (CI step), then `COPY` and `pip install` the built wheel inside the Dockerfile. More invasive but cleanest separation.

In all cases:

- [ ] The `Release` workflow on `main` must pass end-to-end and produce a `ghcr.io/hhopke/intervals-icu-mcp:latest` image.
- [ ] `make docker/build` locally must continue to work (or the docs updated to match).
- [ ] Verify the resulting image's `python -m intervals_icu_mcp --version` (or equivalent) reports a sensible version, not `0.0.0` or unknown.

## Files involved

- `.github/workflows/release.yml` — Build and Push Docker Image job
- `Dockerfile` — multi-stage build that runs `uv sync`
- `pyproject.toml` — `[tool.hatch.version] source = "vcs"` config

## Effort

S (~30-50 LOC change across Dockerfile + release.yml; main risk is iterating on what `hatch-vcs` accepts as a `PRETEND_VERSION` for non-tagged commits)

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -25,6 +25,20 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v6
+        with:
+          fetch-depth: 0
+
+      - name: Set up uv
+        uses: astral-sh/setup-uv@v7
+
+      - name: Compute package version
+        id: version
+        run: |
+          set -euo pipefail
+          uv build --wheel --out-dir /tmp/wheel
+          PKG_VERSION=$(ls /tmp/wheel/intervals_icu_mcp-*.whl | sed -E 's|.*/intervals_icu_mcp-([^-]+)-.*|\1|')
+          echo "pkg_version=$PKG_VERSION" >> "$GITHUB_OUTPUT"
+          echo "Resolved PKG_VERSION=$PKG_VERSION"
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v4
@@ -54,6 +68,8 @@ jobs:
           context: .
           platforms: linux/amd64,linux/arm64
           push: true
+          build-args: |
+            PKG_VERSION=${{ steps.version.outputs.pkg_version }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
           cache-from: type=gha

--- a/Dockerfile
+++ b/Dockerfile
@@ -9,6 +9,12 @@ COPY --from=uv /uv /usr/local/bin/uv
 # Set working directory
 WORKDIR /app
 
+# Version is derived from git via hatch-vcs at build time, but the .git
+# directory is not in the Docker build context. The CI workflow builds a
+# wheel locally to compute the version and passes it through as a build-arg.
+ARG PKG_VERSION=0.0.0+docker
+ENV SETUPTOOLS_SCM_PRETEND_VERSION=$PKG_VERSION
+
 # Copy dependency files and README (needed for package metadata)
 COPY pyproject.toml uv.lock* README.md ./
 

